packages/appium/lib/cli/setup-command.js

```diff
@@ -0,0 +1,163 @@
+import _ from 'lodash';
+import { SETUP_SUBCOMMAND } from '../constants';
+import {runExtensionCommand} from './extension';
+import { fs } from '@appium/support';
+
+/**
+ * Driver names listed in KNOWN_DRIVERS to install by default.
+ */
+export const MOBILE_DRIVERS = ['uiautomator2', 'espresso', 'xcuitest'];
+export const SUBCOMMAND_MOBILE = 'mobile';
+
+
+/**
+ * Driver names listed in KNOWN_DRIVERS to install by default.
+ */
+export const DESKTOP_APP_DRIVERS = ['mac2'];
+export const SUBCOMMAND_DESKTOP = 'desktop';
+
+/**
+ * Driver names listed in KNOWN_DRIVERS to install by default.
+ */
+export const BROWSER_DRIVERS = ['uiautomator2', 'xcuitest', 'chromium', 'safari', 'gecko'];
+export const SUBCOMMAND_BROWSER = 'browser';
+
+/**
+ * Plugin names listed in KNOWN_PLUGINS to install by default.
+ */
+export const DEFAULT_PLUGINS = [];
+
+
+/**
+ * A subcommand to remove ANDROID_HOME
+ */
+export const SUBCOMMAND_RESET = 'reset';
+
+
+/**
+ * Run 'setup' command to install drivers/plugins into the given appium home.
+ * @template {import('appium/types').CliCommandSetup} SetupCmd
+ * @template {import('appium/types').CliExtensionCommand} ExtCmd
+ * @param {string} appiumHome
+ * @param {import('appium/types').Args<SetupCmd>} preConfigArgs
+ * @param {import('../extension/extension-config').ExtensionConfig<ExtCmd>} driverConfig
+ * @param {import('../extension/extension-config').ExtensionConfig<ExtCmd>} pluginConfig
+ * @returns
+ */
+export async function runSetupCommand(appiumHome, preConfigArgs, driverConfig, pluginConfig) {
+  if (preConfigArgs.setupCommand === SUBCOMMAND_RESET) {
+    await fs.rimraf(appiumHome);
+    return;
+  }
+
+  if (!_.isEmpty(driverConfig.installedExtensions) || !_.isEmpty(pluginConfig.installedExtensions)) {
```

Comment:
As stated above instead of throwing an error we may just install missing drivers/plugins